### PR TITLE
docs: add config generation docs in feature

### PR DESCRIPTION
**Pull request by laststylebender14**

### CODE REVIEW

File: docs/config-generation.md [hunk 1]
```diff
@@ -0,0 +1,887 @@
...
+</head>
+
+import Tabs from "@theme/Tabs"
+import TabItem from "@theme/TabItem"
+
+# GraphQL Configuration Generation with Tailcall
+
+## What is Configuration Generation?
+
+Configuration generation is the process of automatically generating graphQL configurations from the various sources such as **REST Endpoints**, **Proto files** and already existing **GraphQL configuration** files.
```
Comment (tusharmath) [inline docs/config-generation.md]: We should say, REST and gRPC instead of proto files.

File: docs/config-generation.md [hunk 2]
```diff
@@ -0,0 +1,887 @@
...
+
+### Problems It Solves
+
+Configuration generation with Tailcall addresses several critical challenges in creating and maintaining GraphQL configurations, ensuring streamlined and error-free processes.
+
+1. **Consistency**: By automatically generating schemas, types etc. Tailcall maintains a consistent structure, making it easier to manage and understand the API.
+2. **Efficiency**:
+   - **Time Savings**: Significantly reduces the time required to generate and update GraphQL configurations by automating the process, allowing developers to focus on core functionalities.
+   - **Quick Migration**: Enables rapid migration from REST or Protbuff to GraphQL, transforming complex API structures into GraphQL schemas within minutes.
```
Comment (tusharmath) [inline docs/config-generation.md]: Both are talking about the same problem.

File: docs/config-generation.md [hunk 3]
```diff
@@ -0,0 +1,887 @@
...
+3. **Error Reduction**:
+   - **Minimized Human Errors**: Automates the tedious and error-prone task of manually writing GraphQL configurations, ensuring accuracy and reducing the risk of mistakes.
+   - **Accurate Type Mapping**: Automatically maps API responses to GraphQL types, ensuring correct and precise type definitions without manual intervention.
```
Comment (tusharmath) [inline docs/config-generation.md]: Both are the same.

File: docs/config-generation.md [hunk 4]
```diff
@@ -0,0 +1,887 @@
...
+
+<head>
+<meta property="og:type" content="article"/>
+  <title>Generate GraphQL Configuration</title>
+</head>
+
+import Tabs from "@theme/Tabs"
+import TabItem from "@theme/TabItem"
+
+# GraphQL Configuration Generation with Tailcall
```
Comment (tusharmath) [inline docs/config-generation.md]: H1 isn't required if title is set above.

File: docs/config-generation.md [hunk 5]
```diff
@@ -0,0 +1,887 @@
...
+   - **Accurate Type Mapping**: Automatically maps API responses to GraphQL types, ensuring correct and precise type definitions without manual intervention.
+
+By addressing these challenges, Tailcall enhances the development workflow, making the process of creating and managing GraphQL configurations more reliable, efficient, and error-free.
+
+### Why is it Hard to Write GraphQL Schemas by Hand?
+
+Writing GraphQL schemas manually presents several challenges that can complicate and slow down the development process:
+
+1. **Complex API Responses**:
+   - **Large and Detailed Responses**: APIs often return extensive and intricate data, making it laborious to map these responses accurately to GraphQL types.
```
Comment (tusharmath) [inline docs/config-generation.md:25]: Someone might have their APIs on Open API or gRPC so why would they need to map those responses?

File: docs/config-generation.md [hunk 6]
```diff
@@ -0,0 +1,887 @@
...
+   - **Nested Structures**: Dealing with deeply nested JSON objects requires careful handling to ensure all relationships and data hierarchies are correctly represented in the schema.
+2. **Data Consistency**:
+   - **Missing Properties**: APIs can have inconsistent data where some items might lack certain properties, necessitating meticulous examination to define accurate types and optional fields.
+   - **Dynamic Data**: Handling APIs with dynamic data fields adds another layer of complexity, requiring flexible and robust schema definitions to accommodate various data shapes.
+3. **Migration Efforts**:
+   - **Manual Workload**: Converting existing REST APIs or Protocol Buffers to GraphQL involves substantial manual effort, as each endpoint must be individually mapped to corresponding GraphQL types and queries.
+   - **Error-Prone Process**: The manual creation of schemas increases the likelihood of errors, leading to potential issues in data fetching and integration.
```
Comment (tusharmath) [inline docs/config-generation.md:36]: This section assumes that the only way to create a GraphQL APIs is by mapping responses to Schema and that's the only challenge. I think there are plenty others — 

- Identify the types and their names (if the type info can't be infereed)
- Merging types that are duplicates
- Optimize configuration to reduce duplication
- Identify types that are used as inputs and outputs.
- Finding joins between APIs (Will be supported soon)

Taking care of so many thing while onboarding thousands of APIs isn't humanly possible hence we equip you with necessary tools to quickly onboard APIs.

File: docs/config-generation.md [hunk 7]
```diff
@@ -0,0 +1,887 @@
...
+
+These challenges highlight the need for automated tools, which streamline the process of generating GraphQL schemas, ensuring accuracy and efficiency while reducing the manual workload and error potential.
+
+For more insights on why manual GraphQL schema writing is becoming obsolete, you can read this [blog post by Tailcall](https://blog.tailcall.run/writing-a-graphql-backend-by-hand-is-long-gone).
+
+## Input Sources
```
Comment (tusharmath) [inline docs/config-generation.md]: This section is better explained below, why have this one here?

File: docs/config-generation.md [hunk 8]
```diff
@@ -0,0 +1,887 @@
...
+  title: String
+  userId: Int
+}
+
+type Query {
+  post(p1: Int!): Post @http(path: "/posts/{{.args.p1}}")
+}
+```
+
+### Effortless Proto Integration
```
Comment (tusharmath) [inline docs/config-generation.md]:
```suggestion
### Effortless gRPC Integration
```

File: docs/config-generation.md (hunk 1 above)
Comment (laststylebender14) [reply docs/config-generation.md]: done.

File: docs/config-generation.md (hunk 2 above)
Comment (laststylebender14) [reply docs/config-generation.md]: removed `problem it solves` section and kept `why is it hard to write graphql schema by hand`

File: docs/config-generation.md (hunk 3 above)
Comment (laststylebender14) [reply docs/config-generation.md]: removed, done.

File: docs/config-generation.md (hunk 4 above)
Comment (laststylebender14) [reply docs/config-generation.md]: done.

File: docs/config-generation.md (hunk 8 above)
Comment (laststylebender14) [reply docs/config-generation.md]: done

File: docs/config-generation.md (hunk 7 above)
Comment (laststylebender14) [reply docs/config-generation.md]: I thought if someone doesn't want scroll all the down to check the input sources which we support. but it's visible on outline as well so we can remove it.

File: docs/config-generation.md (hunk 5 above)
Comment (laststylebender14) [reply docs/config-generation.md:25]: I'm talking about in-general issues that we face.

File: docs/config-generation.md (hunk 6 above)
Comment (laststylebender14) [reply docs/config-generation.md:36]: done, have added some points from above.

**Pull request merged**